portfolio/src/main/java/com/google/sps/servlets/DataServlet.java

```diff
@@ -76,12 +91,14 @@ public void doPost(HttpServletRequest request, HttpServletResponse response) thr
       String userComment = request.getParameter("user-comment");
       long timestamp = System.currentTimeMillis();
       String userEmail = user.getCurrentUser().getEmail();
+      String imageUrl = getUploadedFileUrl(request, "image");
       serverData.add(userComment);
 
       Entity commentEntity = new Entity("Comment");
       commentEntity.setProperty("userComment", userComment);
       commentEntity.setProperty("timestamp", timestamp);
       commentEntity.setProperty("userEmail", userEmail);
+      commentEntity.setProperty("imageUrl", imageUrl);
```

Comment:
Something to think about - getUploadedFileUrl() returns a String, but there are several places where you return null (where you can't get a valid image URL). Does setProperty() handle null property values? If not, you may want to 

1) return the empty string instead of null, and then set the URL to the empty string here, or 
2) check if imageUrl is null before calling setProperty(), and then maybe just not setting this property if it is.

There may be other good solutions, so I'd think a little bit about what works best for you here.

Reply:
I changed it to return an empty string instead of null. Would the code be healthier if I did the other way?

Reply:
I thought about this a bit more. I think it doesn't matter too much whether you use null or an empty string. The important part is that you should handle the case where there is no valid imageUrl. Probably the cleanest way to do it is to have the Javascript check the URL before trying to set an image, and just not include an image if there is no valid URL (or you could display some type of "image not found" message).